plugin/storage/integration/integration.go

```diff
@@ -342,15 +362,27 @@ func (s *StorageIntegration) testFindTraces(t *testing.T) {
 	}
 }
 
-func (s *StorageIntegration) findTracesByQuery(t *testing.T, query *spanstore.TraceQueryParameters, expected []*model.Trace) []*model.Trace {
+func (s *StorageIntegration) findTracesByQuery(t *testing.T, query *tracestore.TraceQueryParams, expected []*model.Trace) []*model.Trace {
 	var traces []*model.Trace
 	found := s.waitForCondition(t, func(t *testing.T) bool {
 		var err error
-		traces, err = s.SpanReader.FindTraces(context.Background(), query)
+		iterTraces := s.TraceReader.FindTraces(context.Background(), *query)
+		tracesFound, err := iter.FlattenWithErrors(iterTraces)
 		if err != nil {
 			t.Log(err)
 			return false
 		}
+
+		for _, traceFound := range tracesFound {
+			modelTrace := &model.Trace{}
```

Comment:
please where can I read more about the contract, chunks, operator.
About the helper function, 
should it be something like this `PTracesSeq2ToModel(iter.Seq2[[]p.trace.Traces, error]) ([]*model.Trace, error)

To be clear, this function should check for the trace id in each chunk(i suppose) from the iterator and group them into respective `model.Trace`

Reply:
The contract is explained in the godocs for the storage API. Yes, that would be the signature of the helper. And yes, it would potentially combine multiple chunks (sequential-only) into single model.Trace